### ISSUE TITLE
Use with recent Rust nightly fails with lockfile version error

### ISSUE THREAD

**Issue by clechasseur** [taiki-e/cargo-minimal-versions#31]
In my projects, I check for MSRV by installing Rust nightly, the minimum Rust version I support and then running the following command-line (more or less):

```sh
cargo minimal-versions check --target-dir target-minimal --workspace --lib --bins --all-features
```

A few days ago, I started seeing failures with a message like this:

```
info: running `cargo hack check --target-dir target-minimal --workspace --lib --bins --all-features`
running `cargo check --target-dir target-minimal --lib --bins --all-features` on mini_exercism (1/1)
  warning: unused manifest key: lints
  error: failed to parse lock file at: /home/runner/work/mini_exercism/mini_exercism/Cargo.lock
  
  Caused by:
    lock file version `4` was found, but this version of Cargo does not understand this lock file, perhaps Cargo needs to be updated?
```

Example: https://github.com/clechasseur/mini_exercism/actions/runs/11097764447/job/30829476960

I tried to run this locally by reverting to the Nightly build from 2024-09-24 and the error disappeared.

I'm not an expert but I _think_ this may be caused by this: https://github.com/rust-lang/cargo/commit/cf781daeb757e0fa73123ebaee9098b3e592ad33

It looks like they switched to a new lockfile version in Nightly, which probably means that the lockfile generated by cargo-minimal-versions is incompatible with pre-1.78 Rust (if I understand correctly).

My questions:
1. Is my interpretation correct?
2. If it is, is this something that I need to fix on my end, or does it require a change in cargo-minimal-versions?

Thanks for this excellent tool btw. 🙂

**Comment by taiki-e** [comment]
Fixed in 0.1.29. Thanks for the report!

**Comment by clechasseur** [comment]
That was fast! Thank you 🔥
